udc/src/main/java/io/crate/udc/ping/PingTask.java

```diff
@@ -64,25 +65,37 @@
     private final ExtendedNodeInfo extendedNodeInfo;
     private final String pingUrl;
     private final Settings settings;
+    private final LicenseService licenseService;
 
     private AtomicLong successCounter = new AtomicLong(0);
     private AtomicLong failCounter = new AtomicLong(0);
 
     public PingTask(ClusterService clusterService,
                     ExtendedNodeInfo extendedNodeInfo,
                     String pingUrl,
-                    ClusterSettings clusterSettings,
-                    Settings settings) {
+                    Settings settings,
+                    LicenseService licenseService) {
         this.clusterService = clusterService;
         this.pingUrl = pingUrl;
         this.settings = settings;
         this.extendedNodeInfo = extendedNodeInfo;
+        this.licenseService = licenseService;
     }
 
     private Map<String, String> getKernelData() {
         return extendedNodeInfo.osInfo().kernelData();
     }
 
+    @VisibleForTesting
+    static String getLicenseExpiryDate(@Nonnull DecryptedLicenseData license) {
+        return String.format("%tF", license.expiryDateInMs());
```

Comment:
Seems also a bit overkill - we're testing the `String.format` function here.

If we wanted to ensure that the pings are sending the information as ISO string we'd have to test on a higher level.